modules/gdscript/language_server/gdscript_extend_parser.cpp

```diff
@@ -488,6 +488,76 @@ String ExtendGDScriptParser::get_text_for_lookup_symbol(const lsp::Position &p_c
 	return longthing;
 }
 
+int ExtendGDScriptParser::get_parameter_count(const lsp::Position &p_cursor, lsp::Position *p_begining_position) {
+
+	int count = 0;
+	p_begining_position->line = p_cursor.line;
+
+	int closest_bracket_opening_pos = lines[p_cursor.line].substr(0, p_cursor.character).find_last("(");
+	int closest_bracket_closing_pos = lines[p_cursor.line].substr(0, p_cursor.character).find_last(")");
+	if (closest_bracket_closing_pos > closest_bracket_opening_pos || closest_bracket_opening_pos == -1) {
+		return 0;
+	}
+	for (int j = closest_bracket_opening_pos; j < p_cursor.character; j++) {
+		if (lines[p_cursor.line][j] == ',') {
+			count++;
+		}
+	}
+	p_begining_position->character = closest_bracket_opening_pos;
+	while (p_begining_position->character > 0 && lines[p_cursor.line][p_begining_position->character] != ' ' && lines[p_cursor.line][p_begining_position->character] != ',') {
+		p_begining_position->character--;
+	}
+	p_begining_position->character++;
+
+	return count;
+}
+
+void ExtendGDScriptParser::get_document_link(List<lsp::DocumentLink> *r_links_parameter) {
+
+	int len = lines.size();
+	String root_uri = GDScriptLanguageProtocol::get_singleton()->get_workspace()->root_uri;
+	for (int i = 0; i < len; ++i) {
+		int pos = 0;
+
+		while (pos < lines[i].size()) {
+
+			int start_pos_1 = lines[i].find("\"", pos);
+			int start_pos_2 = lines[i].find("'", pos);
+			int start_pos;
+			if (start_pos_1 == -1 && start_pos_2 == -1) {
+				break;
+			} else if (start_pos_1 == -1) {
+				start_pos = start_pos_2;
+			} else if (start_pos_2 == -1) {
+				start_pos = start_pos_1;
+			} else {
+				start_pos = MIN(start_pos_1, start_pos_2);
+			}
+
+			int end_pos = lines[i].find(lines[i].substr(start_pos, 1), start_pos + 1);
+			String value = lines[i].substr(start_pos + 1, end_pos - start_pos - 1);
+
+			if (!value.begins_with("res://")) {
+				value = "res://" + value;
```

Comment:
I don't think this is the right way to handle relative file paths.

How about this situation:
res://enermys/wolf.gd
```gdscript
extends 'enermy.gd'
```
`'enermy.gd'` should link to `res://enermys/enermy.gd` instead of `res://enermy.gd`.